pkg/serverless/trigger/extractor.go

```diff
@@ -188,11 +188,20 @@ func GetTagsFromALBTargetGroupRequest(event events.ALBTargetGroupRequest) map[st
 	httpTags := make(map[string]string)
 	httpTags["http.url_details.path"] = event.Path
 	httpTags["http.method"] = event.HTTPMethod
-	if event.Headers != nil {
-		if event.Headers["Referer"] != "" {
-			httpTags["http.referer"] = event.Headers["Referer"]
+
+	headers := event.Headers
+	if headers == nil && event.MultiValueHeaders != nil {
+		headers = make(map[string]string)
+		for k, v := range event.MultiValueHeaders {
+			headers[k] = v[0]
 		}
-		if ua := event.Headers["User-Agent"]; ua != "" {
+	}
```

Comment:
You can do this without an extra allocation by checking `event.MultiValueHeaders["Referer"] != nil && len(event.MultiValueHeaders["Referer"]) > 0`

Reply:
Thank you for the suggestion!
I've implemented this improvement in commit aaaa24f04199be385ee9df4553e9e7b4217a236e.